pkg/util/glog/glog.go

```diff
@@ -1,101 +1,225 @@
+}
+
+func (glogger) Fatal(args ...interface{}) {
+	log.FatalDepth(2, args...)
+}
+
+// gverbose handles glog.V(x) calls.
+type gverbose struct {
+	log.Verbose
 }
 
 func (g gverbose) Infof(format string, args ...interface{}) {
 	g.Verbose.Infof(format, args...)
 }
 
-// file logs the provided messages at level or below to the writer, or delegates
+func (g gverbose) Info(args ...interface{}) {
+	g.Verbose.Info(args...)
+}
+
+// FileLogger logs the provided messages at level or below to the writer, or delegates
 // to glog.
-type file struct {
-	w     io.Writer
-	level int
+type FileLogger struct {
+	w     *bufio.Writer
+	level int32
 }
 
-func (f file) Is(level int) bool {
+// Is returns whether the current logging level is greater than or equal to the parameter.
+func (f *FileLogger) Is(level int32) bool {
 	return level <= f.level
 }
 
-func (f file) V(level int) Logger {
-	// only log things that glog allows
-	if !glog.V(glog.Level(level)) {
-		return None
-	}
-	// send anything above our level to glog
-	if level > f.level {
-		return Log
+// V will returns a logger which will discard output if the specified level is greater than the current logging level.
+func (f *FileLogger) V(level int32) VerboseLogger {
+
+	// Is the loglevel set verbose enough to accept the forthcoming log statement
+	if log.V(log.Level(level)) {
+
+		// If the loglevel has been elevated above this file logger's verbosity (generally set to 2)
+		// then delegate ALL messages to glog in order to leverage its file/line/timestamp
+		// prefix information.
+		if log.V(log.Level(f.level + 1)) {
+			return Log
+		}
+
+		// Otherwise, send enable our file logging implementation to receive the log statement
+		return f
 	}
-	return f
+
+	// Otherwise discard
+	return None
 }
 
-func (f file) Infof(format string, args ...interface{}) {
-	fmt.Fprintf(f.w, format, args...)
-	if !strings.HasSuffix(format, "\n") {
-		fmt.Fprintln(f.w)
+func (f *FileLogger) writeln(line string) {
+	defer f.w.Flush()
+	if f.level >= 100 {
+		pc := make([]uintptr, 10)
+		runtime.Callers(4, pc)
+		function := runtime.FuncForPC(pc[0])
+		file, line := function.FileLine(pc[0])
+		s := strings.Split(file, "/") // remove absolute path from file
+		file = s[len(s)-1]
+		f.w.WriteString(fmt.Sprintf("[%s:%d] ", file, line-1))
 	}
+
+	f.w.WriteString(line)
+	if !strings.HasSuffix(line, "\n") {
+		f.w.WriteByte('\n')
+	}
+}
+
+func (f *FileLogger) outputf(format string, args ...interface{}) {
+	f.writeln(fmt.Sprintf(format, args...))
+}
+
+func (f *FileLogger) output(args ...interface{}) {
+	f.writeln(fmt.Sprint(args...))
+}
+
+// Infof records an info log entry.
+func (f *FileLogger) Infof(format string, args ...interface{}) {
+	f.outputf(format, args...)
+}
+
+// Info records an info log entry.
+func (f *FileLogger) Info(args ...interface{}) {
+	f.output(args...)
+}
+
+// Warningf records an warning log entry.
+func (f *FileLogger) Warningf(format string, args ...interface{}) {
+	f.outputf(format, args...)
+}
```

Comment:
With the delegation model we are zeroing in on, my thought was that it would lead to inconsistency. If a developer wanted to be sure a warning was clear even to a casual user, they would be tempted to do : glog.Warning( "WARNING: something went wrong" )
- When delegating to glog proper, this would result in the expected outcome: "*W* [normal glog prefix] WARNING: something went wrong"
- When the file logger was in operation, however, it would result in "WARNING: WARNING: something when wrong"

If you are liking the prefix idea, I think the options are:
(1) Inject the prefix only in the file logger path, encourage devs to never use their own prefix, and accept that the glog and file logger output will be a little different. I think this reduces the searchability of logs since a developer won't be sure if they are searching for "W " or "WARNING" unless they are aware of this subtlety. 
(2) Inject the prefix when using the file logger *and* prior to handing the message to glog, so log output will always feel consistent (only the presence or absence of a header would be obvious). I hesitate to do this without your input since it seemed to be edging us closer to our own logging implementation.